tests/test_write.py

```diff
@@ -272,3 +273,12 @@ def test_write_noncontiguous(tmpdir):
     with rasterio.open(name, 'w', **kwargs) as dst:
         for i in range(BANDS):
             dst.write(arr[:, :, i], indexes=i + 1)
+
+
+def test_write_blacklist(tmpdir):
```

Comment:
@perrygeo this is decoupled from the list you created above in `_io`.  How about importing and iterating over `_io.BAD_WRITE_DRIVERS` here, so that both are in sync?